transcription/ibm.go

```diff
@@ -0,0 +1,134 @@
+package transcription
+
+import (
+	"bufio"
+	"encoding/base64"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// IBMResult is the result of an IBM transcription. See
+// https://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/doc/speech-to-text/output.shtml
+// for details.
+type IBMResult struct {
+	ResultIndex int              `json:"result_index"`
+	Results     []ibmResultField `json:"results"`
+}
+type ibmResultField struct {
+	Alternatives []ibmAlternativesField `json:"alternatives"`
+	Final        bool                   `json:"final"`
+}
+type ibmAlternativesField struct {
+	WordConfidence    []ibmWordConfidence `json:"word_confidence"`
+	OverallConfidence float64             `json:"confidence"`
+	Transcript        string              `json:"transcript"`
+	Timestamps        []ibmWordTimestamp  `json:"timestamps"`
+}
+type ibmWordConfidence []interface{}
```

Comment:
Can you explain this type? Seems like a confidence is well-defined list of two elements:
>A confidence score for each word of the transcript as a list of lists. Each inner list consists of two elements: the word and its confidence score in the range of 0 to 1. For example, [[\"hello\",0.95],[\"world\",0.866]]. Available only for the best alternative and only in results marked as final.